### ISSUE TITLE
Delete Pad Improvements

### ISSUE THREAD

**Issue by Vringe** [ether/etherpad#6758]
* * *

name: Feature request
about: Improvements for the recently added "Delete Pad" Feature
title: 'Improvements for the recently added "Delete Pad" Feature'
labels: enhancement
assignees: 

* * *

Thank you for adding the "Delete Pad" Button in 2.2.6. I think it would be a nice improvement, if the button is only visible for the owner of that pad. Also, some sort of confirmation message would be nice. Currently, it just disconnects you from the pad and you have to reload manually.

**Comment by Maksim-Burtsev** [comment]
i support the improvement of showing the button only for creator of the pad (maybe with a new setting).

for example in cases where pad created by api, users which use pad see this button but none of them have persmission to delete pad

**Comment by HarshDodiya1** [comment]
@Vringe 
After deleting the pad, the application continues to display the popup shown below instead of navigating to another route. Also upon refreshing the page, the pad becomes editable again. Doesn't this seem like an error?

Do I have to create an issue for this?


![image](https://github.com/user-attachments/assets/c47bd7a5-c4da-4c46-9062-d7156991c811)

**Comment by Vringe** [comment]
@HarshDodiya1 
Do you actually see the contents from before the deletion or just an empty pad? In my case, the pad is deleted and after reloading, you start with a fresh new pad. If you really see the old contents, it makes sense to create a separate issue for this.

My suggestion is to redirect to the start page (where you can create a new pad) and confirm the deletion with a toast message.

**Comment by HarshDodiya1** [comment]
@Vringe Yeah, facing the error, means when I delete the pad, it shows me pop up and when I refresh the page then again I am in the same pad. But I think when we delete the pad we should be redirected to the start page. 

Currently it is this: 

https://github.com/user-attachments/assets/536a2517-5d31-463e-af4d-aba07fa44f8a

As you mention it will be good if we redirect to home page and see some toast of successful deletion.

**Comment by zib-ctrl** [comment]
I would like to control the display of the "Delete pad" button via a global setting / Docker ENV vars.
We use Etherpad in Moodle. So the delete button makes no sense, because it fails anyway, since the pad was created by the API. If the delete function should also work for API-created pads in the future, deleting via this button would make no sense in Moodle (even a redirect to the Etherpad start page or the creation of a new pad would make no sense in this case).

**Comment by JohnMcLear** [comment]
Done :)
